File: app/src/main/java/com/example/intern_anrdoid_2020/QiitaService.java
package com.example.intern_anrdoid_2020;

import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

public interface QiitaService {

    @GET("items")
    @Headers("Content-Type: application/json")
    Call<ArrayList<QiitaArticleResponse>> getQiitaArticle(@Query("page") int page,
                                                          @Query("per_page") int per_page,
                                                          @Query("query") String query);
}

File: app/src/main/java/com/example/intern_anrdoid_2020/QiitaService.kt
package com.example.intern_anrdoid_2020

import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query
import java.util.*

interface QiitaService {
    @GET("items")
    @Headers("Content-Type: application/json")
    fun getQiitaArticle(@Query("page") page: Int,
                        @Query("per_page") per_page: Int,
                        @Query("query") query: String?): Call<ArrayList<QiitaArticleResponse>>
}
